esphome/components/pn7160/__init__.py

```diff
@@ -0,0 +1,238 @@
+            {
+                cv.GenerateID(CONF_TRIGGER_ID): cv.declare_id(nfc.NfcOnTagTrigger),
+            }
+        ),
+        cv.Optional(CONF_DWL_REQ_PIN): pins.gpio_output_pin_schema,
+        cv.Required(CONF_IRQ_PIN): pins.gpio_input_pin_schema,
+        cv.Required(CONF_VEN_PIN): pins.gpio_output_pin_schema,
+        cv.Optional(CONF_WKUP_REQ_PIN): pins.gpio_output_pin_schema,
+        cv.Optional(CONF_EMULATION_MESSAGE): cv.string,
+        cv.Optional(CONF_TAG_TTL): cv.positive_time_period_milliseconds,
+    }
+).extend(cv.COMPONENT_SCHEMA)
+
+
+@automation.register_action(
+    "tag.set_emulation_message",
+    SetEmulationMessageAction,
+    SET_MESSAGE_ACTION_SCHEMA,
+)
+async def pn7160_set_emulation_message_to_code(config, action_id, template_arg, args):
+    paren = await cg.get_variable(config[CONF_ID])
+    var = cg.new_Pvariable(action_id, template_arg, paren)
+    template_ = await cg.templatable(config[CONF_MESSAGE], args, cg.std_string)
+    cg.add(var.set_message(template_))
+    template_ = await cg.templatable(
+        config[CONF_INCLUDE_ANDROID_APP_RECORD], args, cg.bool_
+    )
+    cg.add(var.set_include_android_app_record(template_))
+    return var
+
+
+@automation.register_action(
+    "tag.set_write_message",
+    SetWriteMessageAction,
+    SET_MESSAGE_ACTION_SCHEMA,
+)
+async def pn7160_set_write_message_to_code(config, action_id, template_arg, args):
+    paren = await cg.get_variable(config[CONF_ID])
+    var = cg.new_Pvariable(action_id, template_arg, paren)
+    template_ = await cg.templatable(config[CONF_MESSAGE], args, cg.std_string)
+    cg.add(var.set_message(template_))
+    template_ = await cg.templatable(
+        config[CONF_INCLUDE_ANDROID_APP_RECORD], args, cg.bool_
+    )
+    cg.add(var.set_include_android_app_record(template_))
+    return var
+
+
+@automation.register_action(
+    "tag.emulation_off", EmulationOffAction, SIMPLE_ACTION_SCHEMA
+)
+@automation.register_action("tag.emulation_on", EmulationOnAction, SIMPLE_ACTION_SCHEMA)
+@automation.register_action("tag.polling_off", PollingOffAction, SIMPLE_ACTION_SCHEMA)
+@automation.register_action("tag.polling_on", PollingOnAction, SIMPLE_ACTION_SCHEMA)
+@automation.register_action(
+    "tag.set_clean_mode", SetCleanModeAction, SIMPLE_ACTION_SCHEMA
+)
+@automation.register_action(
+    "tag.set_format_mode", SetFormatModeAction, SIMPLE_ACTION_SCHEMA
+)
+@automation.register_action(
+    "tag.set_read_mode", SetReadModeAction, SIMPLE_ACTION_SCHEMA
+)
+@automation.register_action(
+    "tag.set_write_mode", SetWriteModeAction, SIMPLE_ACTION_SCHEMA
+)
+async def pn7160_simple_action_to_code(config, action_id, template_arg, args):
+    paren = await cg.get_variable(config[CONF_ID])
+    return cg.new_Pvariable(action_id, template_arg, paren)
+
+
+async def setup_pn7160(var, config):
+    await cg.register_component(var, config)
+
+    if CONF_DWL_REQ_PIN in config:
+        pin = await cg.gpio_pin_expression(config[CONF_DWL_REQ_PIN])
```

Comment:
Can we change all of these to the new format? (same for 7150)
```suggestion
    if dwl_req_pin_config := config.get(CONF_DWL_REQ_PIN):
        pin = await cg.gpio_pin_expression(dwl_req_pin_config)
```

Reply:
Done